tactics/eauto.ml

```diff
@@ -41,7 +41,12 @@ let assumption id = e_give_exact (mkVar id)
 
 let e_assumption =
   Proofview.Goal.enter { enter = begin fun gl ->
-    Tacticals.New.tclFIRST (List.map assumption (Tacmach.New.pf_ids_of_hyps gl))
+    (if Flags.version_strictly_greater Flags.V8_6 then
```

Comment:
For `eassumption` I have concrete use cases: an arithmetic lemma whose first premise is `?x <> 0` (I have many hypotheses in my context which match) and the second premise is more complicated and I have only one such hypothesis in my context. I don't want to specify explicitly what is `?x` and I don't want to use a trick such as  `eapply my_lemma; 2: eassumption; assumption`. I just want `eapply my_lemma; eassumption` to work whatever the order of hypotheses.

On the contrary, I can't see any concrete use case to making `assumption` multi-success since it solves the goal and does not affect other goals.